Expected behaviour of --all-scripts flag

## Issue

Given the following:

```
  "scripts": {
    "development": "webpack --config config/webpack.config.development.js",
    "production": "webpack --config config/webpack.config.production.js",
    "validate": "webpack-validator --all-scripts"
  }
```

Running:

`> npm run validate`

Produces the following error:

```

Validating development
Reading: config/webpack.config.development.js
[webpack-validator] Config is valid.
[webpack-validator] Config is valid.
development is valid

Validating production
Reading: config/webpack.config.production.js
[webpack-validator] Config is valid.
[webpack-validator] Config is valid.
production is valid

Validating validate
Reading: webpack.config.js
module.js:442
    throw err;
    ^

Error: Cannot find module 'D:\Dev\webpack-2-spike\webpack.config.js'
```

## Expected
I expected webpack-validator to only execute against the `development` and `production` scripts as they have the `--config` flag.

## Investigation

The issue is that it also parses the `validate` script, which doesn't have a `--config` flag and in this case, the code defaults to `webpack.config.js` which doesn't exist.

See https://github.com/js-dxtools/webpack-validator/blob/b52bf8c298c7da0c9bb4ff493752f1ce29c23bda/src/bin/script-parsers.js#L11

## Question

Is this the expected behavior, or should we ignore scripts that don't point to config files?

--- a/src/bin/script-parsers.js
+++ b/src/bin/script-parsers.js
@@ -1,5 +1,5 @@
 export function webpackConfig(script) {
-  const callsWebpackCli = script.split(' ').includes('webpack')
+  const callsWebpackCli = script.split(' ').indexOf('webpack') !== -1
   if (!callsWebpackCli) return undefined
 
   // Try to parse webpack config name. Not fool proof. Maybe

--- a/src/bin/script-parsers.js
+++ b/src/bin/script-parsers.js
@@ -1,4 +1,7 @@
 export function webpackConfig(script) {
+  const callsWebpackCli = script.split(' ').includes('webpack')
+  if (!callsWebpackCli) return undefined
+
   // Try to parse webpack config name. Not fool proof. Maybe
   // there's a neater way.
   const parts = script.split('--config')

--- a/src/bin/script-parsers.test.js
+++ b/src/bin/script-parsers.test.js
@@ -27,7 +27,6 @@ describe('webpack config name parser', () => {
       'npm run clean-and-copy',
       'webpack-dev-server --env.dev --content-base dist',
       'npm run clean-and-copy',
-      'npm run clean-and-copy',
       'eslint .',
       'npm install && npm run validate',
     ]

--- a/src/bin/script-parsers.test.js
+++ b/src/bin/script-parsers.test.js
@@ -12,6 +12,47 @@ describe('webpack config name parser', () => {
       webpackConfig('webpack --profile --json > stats.json') === 'webpack.config.js'
     )
   })
+
+  it('should return undefined when script does not call the webpack cli', () => {
+    const sciptsThatDoNotCallWebpackCli = [
+      'karma start',
+      'istanbul check-coverage --statements 23 --branches 5 --functions 9 --lines 24',
+      'npm test -- --auto-watch --no-single-run',
+      'npm-run-all --parallel validate-webpack:* lint test --serial check-coverage',
+      'webpack-validator webpack.config.js --env.dev',
+      'webpack-validator webpack.config.js --env.prod',
+      'rimraf dist',
+      'cpy src/index.html src/favicon.ico dist',
+      'npm run clean-dist && npm run copy-files',
+      'npm run clean-and-copy',
+      'webpack-dev-server --env.dev --content-base dist',
+      'npm run clean-and-copy',
+      'npm run clean-and-copy',
+      'eslint .',
+      'npm install && npm run validate',
+    ]
+
+    sciptsThatDoNotCallWebpackCli.forEach(script => {
+      assert(
+        webpackConfig(script) === undefined
+      )
+    })
+  })
+
+  it('should not return undefined when script calls the webpack cli', () => {
+    const scriptsThatCallWebpacCli = [
+      'webpack --env.dev',
+      'webpack --env.prod -p',
+      'NODE_ENV=development webpack --config demo.js',
+      'webpack --profile --json > stats.json',
+    ]
+
+    scriptsThatCallWebpacCli.forEach(script => {
+      assert(
+        webpackConfig(script) !== undefined
+      )
+    })
+  })
 })
 
 describe('node env parser', () => {

--- a/src/bin/validate-all-scripts.js
+++ b/src/bin/validate-all-scripts.js
@@ -6,6 +6,8 @@ module.exports = function validateAllScripts(scripts, quiet) {
     const script = scripts[name]
     const webpackConfigFile = parse.webpackConfig(script)
 
+    if (webpackConfigFile === undefined) return
+
     if (script.indexOf('SET NODE_ENV=') === 0 || script.indexOf('NODE_ENV=') === 0) {
       process.env.node_env = parse.nodeEnv(script)
     } else {

